shared/video-sdk/develop/stream-raw-audio-and-video/project-implementation/windows.mdx

```diff
@@ -326,49 +274,45 @@ To register and use video and audio frame observers in your <Vpl k="CLIENT" />,
-
-                // Set the format of the captured raw audio data.
-                int SAMPLE_RATE = 16000, SAMPLE_NUM_OF_CHANNEL = 1, SAMPLES_PER_CALL = 1024;
+                nRet = mediaEngine->registerVideoFrameObserver(new CustomVideoFrameObserver());
+                nRet = mediaEngine->registerAudioFrameObserver(new CustomAudioFrameObserver());
 
-                agoraEngine->setRecordingAudioFrameParameters(SAMPLE_RATE, SAMPLE_NUM_OF_CHANNEL,
-                    RAW_AUDIO_FRAME_OP_MODE_READ_WRITE, SAMPLES_PER_CALL);
+		        // Set the format of the captured raw audio data.
+		        int SAMPLE_RATE = 16000, SAMPLE_NUM_OF_CHANNEL = 1, SAMPLES_PER_CALL = 1024;
 
-                agoraEngine->setPlaybackAudioFrameParameters(SAMPLE_RATE, SAMPLE_NUM_OF_CHANNEL,
-                    RAW_AUDIO_FRAME_OP_MODE_READ_WRITE, SAMPLES_PER_CALL);
+		        agoraEngine->setRecordingAudioFrameParameters(SAMPLE_RATE, SAMPLE_NUM_OF_CHANNEL,
+                RAW_AUDIO_FRAME_OP_MODE_READ_WRITE, SAMPLES_PER_CALL);
 
-                agoraEngine->setMixedAudioFrameParameters(SAMPLE_RATE, SAMPLE_NUM_OF_CHANNEL, SAMPLES_PER_CALL);
+		        agoraEngine->setPlaybackAudioFrameParameters(SAMPLE_RATE, SAMPLE_NUM_OF_CHANNEL,
+                RAW_AUDIO_FRAME_OP_MODE_READ_WRITE, SAMPLES_PER_CALL);
 
+		        agoraEngine->setMixedAudioFrameParameters(SAMPLE_RATE, SAMPLE_NUM_OF_CHANNEL, SAMPLES_PER_CALL);
             }
-            else {
+            else 
+            {
                 nRet = mediaEngine->registerVideoFrameObserver(NULL);
                 nRet = mediaEngine->registerAudioFrameObserver(NULL);
             }
             return nRet == 0 ? TRUE : FALSE;
         }
         ```
 
-    1. To execute this method before joining the channel, add the following to the `Join` button event listener,
-  `OnBnClickedButton2()`, before `agoraEngine->enableAudio()`:
+    1. To register the observers at startup, in `AgoraImplementationDlg.cpp`, add the following at the of `setupVideoSDKEngine()`:
 
         ```cpp
-        // Enable audio & video frame capture by registering the frames.
+        // Enable and register observers for audio and video frames
         EnableAudioVideoCapture(TRUE);
         ```
-3. **Unregister the video and audio frame observers when you leave a channel**
+3. **Unregister the video and audio frame observers when you close the <Vpl k="CLIENT"/>**
     
-    When you leave a channel, you unregister the frame observers by setting `EnableAudioVideoCapture()` to `false`.
-  To do this, add the following to the`Leave` button event listener, `OnBnClickedButton1()` after
-  `agoraEngine->disableAudio();`:
+    When you quit the <Vpl k="CLIENT"/>, unregister the frame observers by disabling the `EnableAudioVideoCapture()` method with a `false` argument. To implement this workflow, in `AgoraImplementationDlg.cpp`, add the following to `OnClose` before `	agoraEngine->release();`:
```

Comment:
Again the correct place is 'LEAVE' button listener event, not in `OnClose`
Denying these changes.